includes/class-wc-payments-features.php

```diff
@@ -263,6 +275,8 @@ public static function to_array() {
 		return array_filter(
 			[
 				'upe'                     => self::is_upe_enabled(),
+				'upeSplit'                => self::is_upe_split_enabled(),
+				'upeDeferred'             => self::is_upe_deferred_intent_enabled(),
```

Comment:
I suppose the intention here is to supply flag values to the front-end, since [the values are added to the JS script here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/admin/class-wc-payments-admin.php#L539-L543)? If yes, shouldn't we add these two values to [the schema](https://github.com/Automattic/woocommerce-payments/blob/develop/client/globals.d.ts#L10-L16)?

Reply:
My intention here was _not_ actually to supply this to the front-end, but rather to make this available for the unit tests. I wanted to be able to include the UPE flags in [this array](https://github.com/Automattic/woocommerce-payments/blob/add/5682-add-upe-deferred-intent-feature-flag/tests/unit/test-class-wc-payments-features.php#L22-L24), so that I could use the `pre_option_` hook to set these options, as you suggested, and also use [this `tear_down()` function to remove the filters](https://github.com/Automattic/woocommerce-payments/blob/add/5682-add-upe-deferred-intent-feature-flag/tests/unit/test-class-wc-payments-features.php#L42-L45) after use.

However, merely adding the UPE flags to the `FLAG_OPTION_NAME_TO_FRONTEND_KEY_MAPPING` array constant would cause [this test](https://github.com/Automattic/woocommerce-payments/blob/add/5682-add-upe-deferred-intent-feature-flag/tests/unit/test-class-wc-payments-features.php#L64) to fail. We could add them to the front-end schema, though I am a little loathe to do so, since that doesn't seem completely required and many of the other settings returned in this array are not necessarily included in the schema either.

Although, if you did plan on using any of these constants on the front-end, this would be a worthy reason for their inclusion here--so let me know if that might be the case!